Clarify security requirements for blinded claim names (#157)
oauth-wg/oauth-selective-disclosure-jwt: draft-ietf-oauth-selective-disclosure-jwt.md

long enough and has high entropy that it is not practical for the attacker to
guess. A new salt MUST be chosen for each claim.

## Minimum length of the salt {#salt-minlength}

The RECOMMENDED minimum length of the randomly-generated portion of the salt is 128 bits.

Note that minimum 128 bits would be necessary when SHA-256, HMAC-SHA256, or a function of similar strength is used, but a smaller salt size might achieve similar level of security if a stronger iterative derivation function is used.

The issuer MUST ensure that a new salt value is chosen for each claim,

> including for claims that carry the same name at different places in the

Reviewer: ```suggestion
including when the same claim name occurs at different places in the
```

## Holder Binding {#holder_binding_security}
TBD

## Blinding Claim Names {#blinding-claim-names}

It is RECOMMENDED to use cryptographically random numbers with at least 128 bits
of entropy as placeholder claim names.

With the approach chosen in this specification, claim names of objects
that are not themselves selectively disclosable are not blinded.  This
can be seen in Example 6 in the Appendix, where even in the blinded
SD-JWT, `address` and `delivery_address` are visible. This limitation
needs to be taken into account by issuers when creating the structure of
the SD-JWT.

The issuer MUST ensure that a new random placeholder name is chosen for

> each claim, including for claims that carry the same name at different

Reviewer: ```suggestion
each claim, including when the same claim name occurs at different
```
